app/lib/actions/candlepin/environment/set_content.rb

```diff
@@ -2,55 +2,59 @@ module Actions
   module Candlepin
     module Environment
       class SetContent < Candlepin::Abstract
-        input_format do
-          param :cp_environment_id
-          param :content_ids, Array
+        def plan(content_view, environment, content_view_environment, new_content_id = nil)
+          plan_self(:content_view_id => content_view.id,
+                    :environment_id => environment.id,
+                    :cp_environment_id => content_view_environment.cp_id,
+                    :new_content_id => new_content_id)
         end
 
-        def existing_ids
-          ::Katello::Resources::Candlepin::Environment.
-              find(input[:cp_environment_id])[:environmentContent].map do |content|
-            content[:content][:id]
-          end
-        end
-
-        def run
+        def finalize
+          content_view = ::Katello::ContentView.find(input[:content_view_id])
+          environment = ::Katello::KTEnvironment.find(input[:environment_id])
+          content_ids = content_view.repos(environment).map(&:content_id).uniq.compact
+          # in case we create new custom repository that doesn't have the
+          # content_id set yet in the plan phase, we allow to pass it as
+          # additional argument
+          content_ids << input[:new_content_id] if input[:new_content_id] && !content_ids.include?(input[:new_content_id])
```

Comment:
Is this ever actually done?  Grepping for the SetContent action, i never see that passed in.

Reply:
Looks like this was added here: https://github.com/Katello/katello/pull/3812/commits/c0391a38d539047229153d35d38841c4f295d40b but we never used it as such.